File: nuget/IsICloud5GB.Tests/IsiCloud5GbTests.cs
using IsICloud5GB;

namespace IsICloud5GB.Tests;

public class IsiCloud5GbTests
{
    [Fact]
    public void IsICloudFreeStill5Gb_ReturnsTrue()
    {
        Assert.True(IsiCloud5Gb.IsICloudFreeStill5Gb());
    }

    [Fact]
    public void IsICloudFreeStill5Gb_ReturnsBool()
    {
        var result = IsiCloud5Gb.IsICloudFreeStill5Gb();
        Assert.IsType<bool>(result);
    }

    [Fact]
    public void IsICloudFreeStill5Gb_IsIdempotent()
    {
        // No matter how many times you ask, the answer does not change.
        var results = Enumerable.Range(0, 5).Select(_ => IsiCloud5Gb.IsICloudFreeStill5Gb());
        Assert.All(results, r => Assert.True(r));
    }

    [Fact]
    public void IsICloudFreeStill5Gb_IsNotFalse()
    {
        Assert.False(!IsiCloud5Gb.IsICloudFreeStill5Gb());
    }
}

File: spm/Tests/IsICloud5GBTests/IsICloud5GBTests.swift
import XCTest
@testable import IsICloud5GB

final class IsICloud5GBTests: XCTestCase {
    func testIsICloudFreeStill5GB_ReturnsTrue() {
        XCTAssertTrue(IsICloud5GB.isICloudFreeStill5GB())
    }

    func testIsICloudFreeStill5GB_ReturnsBool() {
        let result: Bool = IsICloud5GB.isICloudFreeStill5GB()
        XCTAssertTrue(result)
    }

    func testIsICloudFreeStill5GB_IsIdempotent() {
        // No matter how many times you ask, the answer does not change.
        let results = (0..<5).map { _ in IsICloud5GB.isICloudFreeStill5GB() }
        XCTAssertTrue(results.allSatisfy { $0 })
    }

    func testIsICloudFreeStill5GB_IsNotFalse() {
        XCTAssertFalse(!IsICloud5GB.isICloudFreeStill5GB())
    }
}
